lib/bucky/core/test_core/test_manager.rb

```diff
@@ -34,28 +36,51 @@ def parallel_new_worker_each(data_set, max_processes, &block)
             end
             # Workers decrease when start working
             available_workers -= 1
-            fork { block.call(data) }
+            fork { block.call(data, w_pipe) }
           end
           # Handle all exit code in waitall
           Process.waitall.each do |child|
             Bucky::Core::TestCore::ExitHandler.instance.raise unless child[1].exitstatus.zero?
           end
+
+          w_pipe.close
+          results_set = collect_results_set(r_pipe)
+          r_pipe.close
+
+          results_set
         end
 
         def parallel_distribute_into_workers(data_set, max_processes, &block)
           # Group the data by remainder of index
           data_set_grouped = data_set.group_by.with_index { |_elem, index| index % max_processes }
+          r_pipe, w_pipe = IO.pipe
           # Use 'values' method to get only hash's key into an array
           data_set_grouped.values.each do |data_for_pre_worker|
             # Number of child process is equal to max_processes (or equal to data_set length when data_set length is less than max_processes)
             fork do
-              data_for_pre_worker.each { |data| block.call(data) }
+              data_for_pre_worker.each { |data| block.call(data, w_pipe) }
             end
           end
           # Handle all exit code in waitall
           Process.waitall.each do |child|
             Bucky::Core::TestCore::ExitHandler.instance.raise unless child[1].exitstatus.zero?
           end
+
+          w_pipe.close
+          results_set = collect_results_set(r_pipe)
+          r_pipe.close
+
+          results_set
+        end
+
+        def collect_results_set(r_pipe)
+          results_set = {}
+          r_pipe.each_line do |line|
+            r = JSON.parse(line)
```

Comment:
I think we only need to parse last one line.

example)
```
          last_line = ""
          r_pipe.each_line do |line|
            last_line = line
          end
          r = JSON.parse(last_line)
          results_set[r['test_class_name']] = r
          results_set
```

Reply:
I think need parse every line because of making unique keys.
```
r_pipe.each_line do |line|
  p line
e
end
"{\"test_class_name\":\"Bucky::Core::TestCore::TestClasses::TestServiceAPcE2ePcE2e\",\"cases_count\":1,\"success_count\":0,\"failure_count\":1}\n"
"{\"test_class_name\":\"Bucky::Core::TestCore::TestClasses::TestServiceAPcE2ePcE2e\",\"cases_count\":2,\"success_count\":0,\"failure_count\":2}\n"
"{\"test_class_name\":\"Bucky::Core::TestCore::TestClasses::TestServiceAPcE2eSetupEachPcE2e\",\"cases_count\":1,\"success_count\":0,\"failure_count\":1}\n"
"{\"test_class_name\":\"Bucky::Core::TestCore::TestClasses::TestServiceAPcE2eSetupTeardownEachPcE2e\",\"cases_count\":1,\"success_count\":0,\"failure_count\":1}\n"
"{\"test_class_name\":\"Bucky::Core::TestCore::TestClasses::TestServiceAPcE2eSetupTeardownEachPcE2e\",\"cases_count\":2,\"success_count\":0,\"failure_count\":2}\n"
"{\"test_class_name\":\"Bucky::Core::TestCore::TestClasses::TestServiceAPcE2eTeardownEachPcE2e\",\"cases_count\":1,\"success_count\":0,\"failure_count\":1}\n"
```

Reply:
I see!